### PR TITLE
PR for shrinkwrap/shrinkwrap Umbrella Issue #129

### PR DESCRIPTION
**Pull request by ALRubinger**
Contains fixes for:

#130 
#131 
#132 
#133 
#136

### CODE REVIEW

**Comment by maxandersen** [comment]
i dont know shrinkwrap code base well enough to judge this but +100 on the intent!

**Comment by ALRubinger** [comment]
@maxandersen I think you can judge everything except for #133! Basically - does this build and pass tests in whatever JVMs you've got on your machine that are 8+? And: can we set up some CI builds for these to test against those JVMs going forward?

**Comment by bartoszmajsak** [comment]
> And: can we set up some CI builds for these to test against those JVMs going forward?

I would suggest using circle-ci with several JVMs set up. From my experience, it's way more reliable than Travis these days. I can set it up if that's desired.

Side note: it seems your commits are not matching your gh user :)

**Comment by ALRubinger** [comment]
"I would suggest using circle-ci with several JVMs set up. From my experience, it's way more reliable than Travis these days. I can set it up if that's desired." - Awesome!

"Side note: it seems your commits are not matching your gh user :)" - Hmm, I `cherry-pick`'d these from my corporate Git instance. I may have to manually apply patches using my GitHub user and resubmit.

**Comment by arjantijms** [comment]
Ping @ALRubinger @bartoszmajsak @maxandersen any progress on getting this PR merged?

**Comment by ALRubinger** [comment]
I've put a CircleCI config based on JDK11 here into the `129` branch on my remote. It's added to this PR, and it runs here:

https://app.circleci.com/pipelines/github/ALRubinger/shrinkwrap

@bartoszmajsak Do you know how we can also spin up jobs to ensure this works on JDK8+? I suppose that'd be JDK8 plus every other JDK that isn't EOL'd.

And beyond that, anyone in the JBoss camp with the privileges to get this merged into a new branch called `main`, retire the antiquated `master` branch, and cut a release for the first major version change to a `2.0.0`?

@aslakknutsen @maxandersen

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
+# See: https://circleci.com/docs/2.0/configuration-reference
+version: 2.1
+
+# Define a job to be invoked later in a workflow.
+# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
+jobs:
+  # Below is the definition of your job to build and test your app, you can rename and customize it as you want.
+  build-and-test:
```
Comment (bartoszmajsak) [inline .circleci/config.yml:9]:
```suggestion
  build-and-test:
    parameters:
      jdk-version:
        type: string
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    # These next lines define a Docker executor: https://circleci.com/docs/2.0/executor-types/
+    # You can specify an image from Dockerhub or use one of our Convenience Images from CircleCI's Developer Hub.
+    # Be sure to update the Docker image tag below to openjdk version of your application.
+    # A list of available CircleCI Docker Convenience Images are available here: https://circleci.com/developer/images/image/cimg/openjdk
+    docker:
+      - image: cimg/openjdk:11.0
```
Comment (bartoszmajsak) [inline .circleci/config.yml:15]:
```suggestion
      - image: cimg/openjdk:<< parameters.jdk-version >>
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          path: ~/test-results/junit
+
+
+# Invoke jobs via workflows
+# See: https://circleci.com/docs/2.0/configuration-reference/#workflows
+workflows:
+  shrinkwrap: # This is the name of the workflow, feel free to change it to better match your workflow.
+    # Inside the workflow, you define the jobs you want to run.
+    jobs:
+      - build-and-test
```
Comment (bartoszmajsak) [inline .circleci/config.yml:48]:
```suggestion
      - build-and-test
          name: build-and-test-on-jdk-<< matrix.jdk-version >>
          matrix:
            parameters:
              jdk-version: ["8.0", "11.0"]
```

**Review by bartoszmajsak — COMMENTED** [review]
@ALRubinger try this (you can add these 3 suggestions as one batch commit to your branch)

**Comment by bartoszmajsak** [comment]
> And beyond that, anyone in the JBoss camp with the privileges to get this merged into a new branch called `main`, retire the antiquated `master` branch, and cut a release for the first major version change to a `2.0.0`?

@ALRubinger Unless there is some automation relying on GH APIs the rename from `master` -> `main` should be fairly straightforward. I learned the hard way that with REST or GraphQL calls this is not working seamlessly. Or do you mean branching out with `main` from some other point than where the current `master` is?

**Comment by ALRubinger** [comment]
Brilliant as always @bartoszmajsak !!!

Thank you!

image

This also caught that we're not yet working in when run on JRE 17.

Opened #136 and will address that in this PR when done as well.

JBoss folks: I have permissions to do the rename of the `master` branch to `main` and also push these commits up there. 

1) Is there no existing maintainer at JBoss who would like to do that push?
2) Once we get that done, any volunteer with JBoss privileges to cut a release and get it into the JBoss and Maven Central repositories?

If no, we can discuss asking JBoss/Red Hat/IBM if perhaps we'd like to move ShrinkWrap into a central place where we can cut more releases.

**Comment by bartoszmajsak** [comment]
@ALRubinger I am not aware of any active maintainer. I have the rights so if needed I can push the release button. Just ping me.

**Comment by ALRubinger** [comment]
Thank you, bud.

I'll get #136 into this PR and merge it all into a new `main`, then ping you with a request for verification and a `2.0.0` release.

**Comment by ALRubinger** [comment]
Merged into `upstream/main`. 

Set `main` to be the default branch in GitHub.

Removed `upstream/master`.

Tests running clean on JDK8, 11, and 17:

image

Also verified on the new `main` in my remote:

image

@bartoszmajsak - you're a hero. Would you mind verifying my work?

And if it looks good:

Let's CUT THAT RELEASE!

**Comment by ALRubinger** [comment]
WAIT please. 

I need to rewrite some commits to be from my real user. :P

**Comment by ALRubinger** [comment]
OK all clear sorry about that. :)

image

**Comment by bartoszmajsak-test** [comment]
> Ping @ALRubinger @bartoszmajsak @maxandersen any progress on getting this PR merged?

@arjantijms 2.0.0-beta-1 is out and up for grabs from JBoss repository.  Hopefully soon to be available in Maven Central.

**Comment by bartoszmajsak** [comment]
> Ping @ALRubinger @bartoszmajsak @maxandersen any progress on getting this PR merged?

@arjantijms 2.0.0-beta-1 is out and up for grabs from JBoss repository.  Hopefully soon to be available in Maven Central.

**Comment by arjantijms** [comment]
@bartoszmajsak amazing work, thanks!

**Pull request closed without merging**